fetch.js

```diff
@@ -195,44 +187,42 @@
       }
     }
 
-    if (support.blob) {
-      this.blob = function() {
-        var rejected = consumed(this)
-        if (rejected) {
-          return rejected
-        }
 
-        if (this._bodyBlob) {
-          return Promise.resolve(this._bodyBlob)
-        } else if (this._bodyFormData) {
-          throw new Error('could not read FormData body as blob')
-        } else {
-          return Promise.resolve(new Blob([this._bodyText]))
-        }
+    this.blob = function() {
+      var rejected = consumed(this)
+      if (rejected) {
+        return rejected
       }
 
-      this.arrayBuffer = function() {
-        return this.blob().then(readBlobAsArrayBuffer)
+      if (this._bodyBlob) {
+        return Promise.resolve(this._bodyBlob)
+      } else if (this._bodyFormData) {
+        throw new Error('could not read FormData body as blob')
+      } else {
+        try {
+          return Promise.resolve(new Blob([this._bodyText]))
+        } catch(e) {
+          throw new Error('Failed to construct a Blob, Include a Blob polyfill')
```

Comment:
I don't like this being a catch-all block. It might swallow exceptions that we didn't intend. Instead, simply leave `Promise.resolve(new Blob(...))` unwrapped and let `new Blob` raise a "'Blob' is undefined" ReferenceError.